packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -3211,4 +3211,247 @@ describe("Teal Opcodes", function () {
+      stack.push(stringToBytes("andres"));
+      stack.push(stringToBytes("messi"));
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.INVALID_TYPE
+      );
+    });
+  });
+
+  describe("Gtxns and Gtxnsa", () => {
+    let stack: Stack<StackElem>;
+    let interpreter: Interpreter;
+    let tx0: EncodedTx, tx1: EncodedTx;
+
+    this.beforeAll(() => {
+      interpreter = new Interpreter();
+      interpreter.runtime = new Runtime([]);
+      interpreter.tealVersion = MaxTEALVersion;
+      tx0 = TXN_OBJ;
+      tx1 = { ...tx0, fee: 1011, amt: 2300, apaa: ["argA", "argB", "argC"].map(Buffer.from) };
+      interpreter.runtime.ctx.gtxs = [tx0, tx1];
+    });
+
+    this.beforeEach(() => { stack = new Stack<StackElem>(); });
+
+    it("Gtxns: should push value of txfield from tx in group", () => {
+      stack.push(0n); // tx to fetch "fee" of (set as first)
+      let op = new Gtxns(["Fee"], 1, interpreter);
+      op.execute(stack);
+      assert.equal(1, stack.length());
+      assert.equal(BigInt(tx0.fee), stack.pop());
+
+      stack.push(0n);
+      op = new Gtxns(["Amount"], 1, interpreter);
+      op.execute(stack);
+      assert.equal(1, stack.length());
+      assert.equal(BigInt(tx0.amt), stack.pop());
+
+      stack.push(1n); // should fetch data from 2nd tx in group
+      op = new Gtxns(["Fee"], 1, interpreter);
+      op.execute(stack);
+      assert.equal(1, stack.length());
+      assert.equal(BigInt(tx1.fee), stack.pop());
+
+      stack.push(1n);
+      op = new Gtxns(["Amount"], 1, interpreter);
+      op.execute(stack);
+      assert.equal(1, stack.length());
+      assert.equal(BigInt(tx1.amt), stack.pop());
+
+      // gtxn, gtxns also accepts array fields
+      stack.push(1n);
+      op = new Gtxns(["ApplicationArgs", "2"], 1, interpreter);
+      op.execute(stack);
+      assert.equal(1, stack.length());
+      assert.deepEqual(stringToBytes("argC"), stack.pop());
+    });
+
+    it("Gtxns: should panic if length of stack is < 1", () => {
+      const op = new Gtxns(["Fee"], 1, interpreter);
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.ASSERT_STACK_LENGTH
+      );
+    });
+
+    it("Gtxns: should panic if transaction index is out of bounds", () => {
+      stack.push(5n); // we only have 2 transactions in group
+      const op = new Gtxns(["Fee"], 1, interpreter);
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.INDEX_OUT_OF_BOUND
+      );
+    });
+
+    it("Gtxnsa: should push value of txfieldArr[index] from tx in group", () => {
+      stack.push(0n);
+      let op = new Gtxnsa(["ApplicationArgs", "1"], 1, interpreter);
+      op.execute(stack);
+      assert.equal(1, stack.length());
+      assert.deepEqual(stringToBytes("arg2"), stack.pop()); // args from tx0
+
+      stack.push(1n);
+      op = new Gtxnsa(["ApplicationArgs", "0"], 1, interpreter);
```

Comment:
this seems same as `Gtxns` case. what is the difference ?

Reply:
no difference, array fields are supported in both `Gtxns` & `Gtxnsa`, so added test for both.